File: app/src/main/java/ykk/cb/com/cbwms/produce/adapter/Prod_WorkBySaoMa_SearchDetailAdapter.java
package ykk.cb.com.cbwms.produce.adapter;

import android.app.Activity;
import android.text.Html;
import android.widget.TextView;

import java.text.DecimalFormat;
import java.util.List;

import ykk.cb.com.cbwms.R;
import ykk.cb.com.cbwms.model.WorkRecordSaoMaEntry1;
import ykk.cb.com.cbwms.util.basehelper.BaseArrayRecyclerAdapter;

public class Prod_WorkBySaoMa_SearchDetailAdapter extends BaseArrayRecyclerAdapter<WorkRecordSaoMaEntry1> {
    private DecimalFormat df = new DecimalFormat("#.######");
    private Activity context;
    private MyCallBack callBack;

    public Prod_WorkBySaoMa_SearchDetailAdapter(Activity context, List<WorkRecordSaoMaEntry1> datas) {
        super(datas);
        this.context = context;
    }

    @Override
    public int bindView(int viewtype) {
        return R.layout.prod_work_saoma_search_fragment1_detail_item;
    }

    @Override
    public void onBindHoder(RecyclerHolder holder, final WorkRecordSaoMaEntry1 entity, final int pos) {
        // 初始化id
        TextView tv_row = holder.obtainView(R.id.tv_row);
        TextView tv_process = holder.obtainView(R.id.tv_process);
        TextView tv_num = holder.obtainView(R.id.tv_num);
        TextView tv_barcode = holder.obtainView(R.id.tv_barcode);
        TextView tv_reportWay = holder.obtainView(R.id.tv_reportWay);

        // 赋值
        tv_row.setText(String.valueOf(pos + 1));
        tv_process.setText(entity.getProcedureName());
        if(entity.getStrLocationQty().indexOf(":") > -1) {
            tv_num.setText(Html.fromHtml("<small>"+entity.getStrLocationQty()+"</small>"));
        } else {
            tv_num.setText(Html.fromHtml(entity.getStrLocationQty()));
        }
        tv_barcode.setText(entity.getBarcode());
        // 工序汇报方式( A:自动汇报  B:手工汇报 )
        if(entity.getReportWay() == 'A') {
            tv_reportWay.setText("自动汇报");
        } else {
            tv_reportWay.setText("手工汇报");
        }
    }

    public void setCallBack(MyCallBack callBack) {
        this.callBack = callBack;
    }

    public interface MyCallBack {
//        void onClick_num(WorkRecordSaoMaEntry1 entity, int position);
    }


}

File: app/src/main/java/ykk/cb/com/cbwms/produce/adapter/Prod_WorkBySaoMa_SearchDetailAdapter.kt
package ykk.cb.com.cbwms.produce.adapter

import android.app.Activity
import android.text.Html
import android.widget.TextView
import ykk.cb.com.cbwms.R
import ykk.cb.com.cbwms.model.WorkRecordSaoMaEntry1
import ykk.cb.com.cbwms.util.basehelper.BaseArrayRecyclerAdapter
import ykk.cb.com.cbwms.util.basehelper.BaseRecyclerAdapter
import java.text.DecimalFormat

class Prod_WorkBySaoMa_SearchDetailAdapter(private val context: Activity, datas: List<WorkRecordSaoMaEntry1>) : BaseArrayRecyclerAdapter<WorkRecordSaoMaEntry1>(datas) {
    private val df = DecimalFormat("#.######")
    private var callBack: MyCallBack? = null

    override fun bindView(viewtype: Int): Int {
        return R.layout.prod_work_saoma_search_fragment1_detail_item
    }

    override fun onBindHoder(holder: BaseRecyclerAdapter.RecyclerHolder, entity: WorkRecordSaoMaEntry1, pos: Int) {
        // 初始化id
        val tv_row = holder.obtainView(R.id.tv_row) as TextView
        val tv_process = holder.obtainView(R.id.tv_process) as TextView
        val tv_num = holder.obtainView(R.id.tv_num) as TextView
        val tv_barcode = holder.obtainView(R.id.tv_barcode) as TextView
        val tv_reportWay = holder.obtainView(R.id.tv_reportWay) as TextView

        // 赋值
        tv_row.text = (pos + 1).toString()
        tv_process.text = entity.procedureName
        if (entity.strLocationQty.indexOf(":") > -1) {
            tv_num.text = Html.fromHtml("<small>" + entity.strLocationQty + "</small>")
        } else {
            tv_num.text = Html.fromHtml(entity.strLocationQty)
        }
        tv_barcode.text = entity.barcode
        // 工序汇报方式( A:自动汇报  B:手工汇报 )
        if (entity.reportWay == 'A') {
            tv_reportWay.text = "自动汇报"
        } else {
            tv_reportWay.text = "手工汇报"
        }
    }

    fun setCallBack(callBack: MyCallBack) {
        this.callBack = callBack
    }

    interface MyCallBack//        void onClick_num(WorkRecordSaoMaEntry1 entity, int position);


}
